fineract-provider/src/main/java/org/apache/fineract/infrastructure/codes/service/CodeValueReadPlatformServiceImpl.java

```diff
@@ -96,12 +97,27 @@ public CodeValueData retrieveCodeValue(final Long codeValueId) {
             this.context.authenticatedUser();
 
             final CodeValueDataMapper rm = new CodeValueDataMapper();
-            final String sql = "select " + rm.schema() + "where cv.id = ? order by position";
+            final String sql = "select " + rm.schema() + " where cv.id = ? and cv.is_active = 1 order by position";
 
             return this.jdbcTemplate.queryForObject(sql, rm, new Object[] { codeValueId });
         } catch (final EmptyResultDataAccessException e) {
             throw new CodeValueNotFoundException(codeValueId);
         }
 
     }
+
+    @Override
+    public Collection<CodeValueData> retrieveAllActiveCodeValues(Long codeId) {
+        final Collection<CodeValueData> codeValues = this.retrieveAllCodeValues(codeId);
```

Comment:
Suggestion: get Active Code values from DB instead of getting all and looping them to find active ones.